std.conv: add writeText, writeWText, writeDText

pbackus opened the pull request:
These are variants of text, wtext, and dtext that write their output to an output range instead of returning a string.

Fixes #10550

pbackus commented:
Style check is going to fail because of https://github.com/dlang-community/D-Scanner/issues/956

pbackus commented:
CC @LightBender

LightBender commented:
Naming nit: 
writeWtext = writeW**T**ext
writeDtext = writeD**T**ext

pbackus commented:
The current names were chosen to be consistent with `text`, `wtext`, and `dtext`, which do not capitalize the word "text." I don't mind changing them if we think quality of new names is more important than consistency with old ones, though.

LightBender commented:
In v3 those methods are getting renamed. Likely to something like `asText`, `asWText`, and `asDText`. There is a thread about these names in the v3 design repo.

0xEAB commented:
Backup of the original branch: <https://github.com/0xEAB/phobos/tree/fix-gh-10550-backup>

pbackus commented:
@0xEAB Is there a reason you've force-pushed on top of my PR?

pbackus commented:
Ok, looks like you've rebased onto current master. Next time, please just notify me so that I can do it myself. I actually have an unpublished commit on this branch (adding a changelog entry), and now I'm going to have to clean up your mess in order to push it.

pbackus commented:
Rebased, added changelog, and disabled "Allow edits by maintainers," since apparently that's a privilege that not everyone here can be trusted with. Ping me if you need it re-enabled.

thewilsonator commented on std/conv.d:
```diff
@@ -5076,6 +5079,75 @@ if (T.length > 0) { return textImpl!dstring(args); }
+
+    const color = "red";
+    auto output = appender!string();
+    output.writeText(i"My favorite color is $(color)");
```
Looks like dscanner doesn't handle istrings yet?
```
Running DScanner
generated/dscanner-5a53c538d0aa832f03840840271b6631fbbfc53d/dsc --config .dscanner.ini -I ../dmd/druntime/src/ --styleCheck etc std -I.
std/conv.d(5121:23)[error]: Expected `)` instead of stringLiteral `"My favorite color is $(color)"`
std/conv.d(5121:55)[warn]: Empty declaration
```

pbackus replied:
Yes, this is blocked by https://github.com/dlang-community/D-Scanner/pull/965

LightBender replied:
This PR has since been merged. Can we get this over the finish line?

pbackus replied:
Still needs https://github.com/dlang/phobos/pull/10720

0xEAB commented:
> Ok, looks like you've rebased onto current master. Next time, please just notify me so that I can do it myself.

Sorry, will do so.

> Rebased, added changelog, and disabled "Allow edits by maintainers," since apparently that's a privilege that not everyone here can be trusted with.

- I was curious whether the D-Scanner issue was still a thing, given that at least libdparse had been updated.
- Also, another PR of mine had caused a conflict with this one already.
- Given that this PR looked quite “finished” (and was suffering only from the external style checking incompatibility), I figured you wouldn’t mind having it rebased for you. I’m sorry.

pbackus commented:
Rebased again to include https://github.com/dlang/phobos/pull/10720
